rfcs/007 - Peer review infra changes.md

```diff
@@ -0,0 +1,125 @@
+
+## Summary
+
+Eng teams move over from using cloud UIs to IaC because cloud infrastructure changes can lead to outages more often than business logic changes. As a result, it is considered a DevOps best practice to code review infrastructure changes using IaC tools and show the `preview` of the resulting change in pull requests to raise the quality bar and reduce the possibility of mistakes. Cloud infrastructure cannot be code reviewed with IaSQL using migration systems even if they support non-schema, or data, migrations (some ORMs like Prisma only support schema changes), and each migration is wrapped in an IaSQL transaction. Through our research and dogfood efforts, we have not found a migration system that fulfills the requirements needed to code review changes with IaSQL:
+- Allow data-only, `up` migrations
+- Allow plain SQL or have an introspective ORM
+- Free and ideally open source
+- Baseline/ignore certain migrations
+- Be resilient to `iasql_ugprade`
+
+Data migrations require being able to baseline migrations to avoid duplicate record violations on `iasql_upgrade`. Flyway comes closest to meeting all our requirements. However, it is not possible to baseline/ignore existing migrations on their free product.
+
+We could configure `iasql_upgrade` to preserve a `migrations` table and make the name of the table configurable, but the baselining issue would remain as `iasql_upgrade` can change the schema which renders existing migrations on past IaSQL versions invalid. As a result, we need an opinionated solution to unblock peer review of infra changes.
+
+## Proposal
+
+Take inspiration from [RFC 006](./006%20-%20Replicate%20changes%20between%20staging%20and%20prod%20RFC.md) and create a how-to guide in the documentation much like the guide on [how to replicate changes across infra environments](https://iasql.com/docs/envs/) along with an example repository that walks users through how to code review their changes using **convention over configuration** such that the workflow:
+- works with any cloud identity and access management structure
+- works with `iasql_upgrade`
+- works with any source control solution without the need for special access to it
+- preserves the ability to `iasql_preview` changes in development and get the type-safety feedback from the dashboard editor
+
+The workflow simply suggests users run `iasql_begin`, perform all the necessary changes, and before doing `iasql_commit` they manually put the output of `iasql_preview` and the SQL generated thus far into a markdown file within a repository of their liking. Once the PR is approved by their peers and landed, they can manually run `iasql_commit`. We can require a parameter for `iasql_commit(message)` which is a string attached as metadata for the transaction that can be set to be the PR URL. Best practices are thus enforced using convention and not configuration with a complete reference of what was done via the IaSQL audit log if necessary.
```

Comment:
Why call out "markdown file" here? It could just as well be a CSV file since the `iasql_preview` output is a table. I see below a mention of adding an `get_sql_for_transaction` function but it's not mentioned as something to include in the artifact to review. That could be a justification for markdown vs other formats since you can use Github's automatic pretty printing of that file format to your advantage during review, but it's not justified here. 

Also, this flow as described doesn't mention anything about showing the relevant SQL statements but maybe it should? In that case, it may make sense to rework this and the following paragraph.